subprojects/messaging/src/main/java/org/gradle/internal/dispatch/ReflectionDispatch.java

```diff
@@ -33,6 +34,8 @@ public void dispatch(MethodInvocation message) {
             Method method = message.getMethod();
             method.setAccessible(true);
             method.invoke(target, message.getArguments());
+        } catch (AccessControlException e) {
```

Comment:
I don't think this is actually doing anything is it?

Reply:
It's caught here: https://github.com/gradle/gradle/blob/7c7865fe8a43097a0d20ce85afe2a4cd7f0533a4/subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestWorker.java#L111 so that the worker process wouldn't wait forever.

Reply:
Right, but I think `UncheckedException` just re-throws it because its an instance of `RuntimeException`.  I don't think we need to catch/throw it here at all.

Reply:
Yes, you're right. I didn't see the `UncheckedException` code. It's removed now.